restli-server/src/main/java/com/linkedin/restli/internal/server/response/ResponseUtils.java

```diff
@@ -148,7 +148,8 @@ public static RestException buildRestException(RestLiResponseException restLiRes
     }
 
     RestResponse restResponse = responseBuilder.build();
-    return new RestException(restResponse, restLiResponseException.getCause());
+    Throwable cause = restLiResponseException.getCause();
+    return new RestException(restResponse, cause==null ? null : cause.toString(), cause, writableStackTrace);
```

Comment:
Shouldn't the cause.toString() end put printing the entire stack trace? You probably just want to include the exception message for 'message' parameter (cause.getMessage())

Reply:
public Throwable(Throwable cause) {
        fillInStackTrace();
        detailMessage = (cause==null ? null : cause.toString());
        this.cause = cause;
    }

I am thinking of keep the message the same as before and per throwable that is use cause.toString()

Reply:
For RestLiServiceException, the toString is overritten and not include stacktrace.